invoice/src/main/java/org/killbill/billing/invoice/usage/SubscriptionUsageInArrear.java

```diff
@@ -160,26 +174,39 @@ public UsageKey apply(final Usage input) {
                 ContiguousIntervalUsageInArrear existingInterval = inFlightInArrearUsageIntervals.get(usageKey);
                 if (existingInterval == null) {
                     existingInterval = usage.getUsageType() == UsageType.CAPACITY ?
-                                       new ContiguousIntervalCapacityUsageInArrear(usage, accountId, invoiceId, rawSubscriptionUsage, existingTrackingIds, targetDate, rawUsageStartDate, usageDetailMode, internalTenantContext) :
-                                       new ContiguousIntervalConsumableUsageInArrear(usage, accountId, invoiceId, rawSubscriptionUsage, existingTrackingIds, targetDate, rawUsageStartDate, usageDetailMode, internalTenantContext);
+                                       new ContiguousIntervalCapacityUsageInArrear(usage, accountId, invoiceId, rawSubscriptionUsage, existingTrackingIds, targetDate, rawUsageStartDate, usageDetailMode, invoiceConfig, internalTenantContext) :
+                                       new ContiguousIntervalConsumableUsageInArrear(usage, accountId, invoiceId, rawSubscriptionUsage, existingTrackingIds, targetDate, rawUsageStartDate, usageDetailMode, invoiceConfig, internalTenantContext);
 
                     inFlightInArrearUsageIntervals.put(usageKey, existingInterval);
                 }
                 // Add billing event for that usage interval
                 existingInterval.addBillingEvent(event);
                 // Remove usage interval for toBeClosed set
                 toBeClosed.remove(usageKey);
+
+                allSeenUnitTypesForBillingEvent.addAll(existingInterval.getUnitTypes());
+                contiguousIntervalsUsageInArrear.add(existingInterval);
+            }
+
+            // Add all seen unit types (across all intervals) for all intervals
+            for (final ContiguousIntervalUsageInArrear contiguousIntervalUsageInArrear : contiguousIntervalsUsageInArrear) {
+                contiguousIntervalUsageInArrear.addAllSeenUnitTypesForBillingEvent(event, allSeenUnitTypesForBillingEvent);
             }
 
             // Build the usage interval that are no longer referenced
             for (final UsageKey usageKey : toBeClosed) {
                 final ContiguousIntervalUsageInArrear interval = inFlightInArrearUsageIntervals.remove(usageKey);
                 if (interval != null) {
                     interval.addBillingEvent(event);
+                    // No usage section for CANCEL events
+                    interval.addAllSeenUnitTypesForBillingEvent(event, usages.isEmpty() ? allSeenUnitTypesForPrevBillingEvent : allSeenUnitTypesForBillingEvent);
```

Comment:
Confused about that one.. I would expect that we pass an empty set for any cancelation, ... I would also expect that for such events `usages` is empty (all the time)